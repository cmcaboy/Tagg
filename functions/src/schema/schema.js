"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var graphql_tools_1 = require("graphql-tools");
var typeDefs_1 = __importDefault(require("./typeDefs"));
var resolvers_1 = __importDefault(require("./resolvers"));
exports.default = graphql_tools_1.makeExecutableSchema({
    typeDefs: typeDefs_1.default,
    resolvers: resolvers_1.default
});
