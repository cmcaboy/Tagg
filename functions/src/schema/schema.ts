import { makeExecutableSchema } from 'graphql-tools';

import {RootQueryType} from './types/root_query_type';
import {mutation} from './mutations';

import typeDefs from './typeDefs';
import resolvers from './resolvers';

export default makeExecutableSchema({
  typeDefs,
  resolvers
});
