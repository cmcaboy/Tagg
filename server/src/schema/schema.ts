import typeDefs from './typeDefs';
import { resolvers } from './resolvers/index';
import neoAPI from './datasources/neo';
import firestoreAPI from './datasources/firestore';
import { driver } from '../db/neo4j';
import { db } from '../db/firestore';

// import statement did not work here
const { ApolloServer } = require('apollo-server-express');

const playground: any = {
  settings: {
    'editor.cursorShape': 'line',
  },
};

// Create a neo4j conneciton once rather than for every request
const session = driver.session();

// set up the data sources our resolvers need
const dataSources = () => ({
  neoAPI: new neoAPI({ session }),
  firestoreAPI: new firestoreAPI({ db }),
});

// context is information shared with all resolvers.
// Here, I will pull out the header info from the request
// and make sure the user is a valid user. Later, I will
// use the roles parameter to validate the user has permission
// to execute a specific query
const context = async ({ req }: { req: any }) => {
  const auth = (req.headers && req.headers.authorization) || '';
  console.log('auth: ', auth);
  // const email = new Buffer(auth, 'base64').toString('ascii');
  // console.log('email: ', email);

  // I am not doing any type of encryption at the moment;
  const email = auth;

  // I could check for email validity here

  // Retrieve user from db
  let neoRaw;
  try {
    neoRaw = await session.run(
      `MATCH (a:User{id:'${email}'}) RETURN a.id, a.email, a.token, a.roles`,
    );
  } catch (e) {
    // I could throw a real error here
    console.log(`Error retreiving user ${email} from database: ${e}`);
    return null;
  }
  const user = {
    id: neoRaw.records[0]._fields[0],
    email: neoRaw.records[0]._fields[1],
    token: neoRaw.records[0]._fields[2],
    roles: neoRaw.records[0]._fields[3],
  };

  return { user };
};

export default new ApolloServer({
  typeDefs,
  resolvers,
  playground,
  context,
  dataSources,
  engine: {
    apiKey: 'service:cmcaboy-2497:fJtoyV5uQQfIQ0I11WiXqg',
  },
});
