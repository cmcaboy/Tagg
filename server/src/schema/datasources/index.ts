const { DataSource } = require("apollo-datasource");

export default class neoAPI extends DataSource {
  constructor({ driver }: { driver: any }) {
    console.log("neoAPI constructor");
    super();

    this.session = driver.session();
  }

  initialize(config: any) {
    this.context = config.context;
  }

  findUser({ id, hostId }: { id: string; hostId: string }) {
    if (id) {
      return this.session
        .run(`Match (n:User {id: '${id}'}) RETURN n`)
        .then((result: any) => result.records)
        .then((records: any) => {
          console.log("records: ", records);
          if (!records.length) {
            return null;
          }
          const properties = records[0]._fields[0].properties;
          return {
            ...properties,
            profilePic: !!properties.pics ? properties.pics[0] : null,
            hostId: !!hostId ? hostId : null
          };
        })
        .catch((e: string) => console.log("id lookup error: ", e));
    } else {
      console.log("Error, no id inputted!");
      return null;
    }
  }
}
