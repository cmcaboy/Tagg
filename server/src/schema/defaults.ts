import { getCurrentDateNeo } from "../middleware/format";

export const newUserDefaults = {
  active: true,
  gender: "female",
  age: 25,
  description: "",
  school: "",
  work: "",
  sendNotifications: true,
  distance: 50,
  token: "",
  latitude: 40.0,
  longitude: -75.0,
  registerDateTime: getCurrentDateNeo(),
  minAgePreference: 18,
  maxAgePreference: 34,
  followerDisplay: "",
  objectionable: false,
  viewObjectionable: true,
  pics: []
};
