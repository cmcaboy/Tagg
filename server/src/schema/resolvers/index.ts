import { Resolvers } from "../../types/generated";
import { merge } from "lodash";

import { Query } from "./Query";
import { User } from "./User";
import { DateBid } from "./DateBid";
import { DateItem } from "./DateItem";
import { Match } from "./Match";
import { Mutation } from "./Mutation";
import { Subscription } from "./Subscription";
import { DateBidList } from "./DateBidList";
import { DateList } from "./DateList";
import { Following } from "./Following";
import { MatchList } from "./MatchList";
import { Message } from "./Message";
import { MessageItem } from "./MessageItem";
import { Queue } from "./Queue";

export const resolvers: Resolvers = {
  DateBid,
  DateBidList,
  DateItem,
  DateList,
  Following,
  Match,
  MatchList,
  Message,
  MessageItem,
  Mutation,
  Query,
  Queue,
  Subscription,
  User
};
