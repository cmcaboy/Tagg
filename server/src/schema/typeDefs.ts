const typeDefs = `
    type User {
        id: String!
        active: Boolean
        name: String
        email: String
        age: Int
        description: String
        school: String
        work: String
        sendNotifications: Boolean
        gender: String
        distance: Int
        token: String
        latitude: Float
        longitude: Float
        minAgePreference: Int
        maxAgePreference: Int
        followerDisplay: String
        match: Boolean
        distanceApart: Float
        order: Float
        registerDateTime: String
        pics: [String]
        profilePic: String
        hasDateOpen: Boolean
        isFollowing: Boolean
        following: Following
        bids: DateBidList
        dateRequests: DateList
        queue: Queue
        matchedDates: MatchList
    }

    type DateList {
        id: String
        list: [DateItem]
        cursor: Float
    }

    type DateBidList {
        id: String
        list: [DateBid]
        cursor: Float
    }

    type Following {
        id: String
        list: [User]
        cursor: Float
    }

    type Queue {
        id: String
        list: [User]
        cursor: Float
    }

    type MatchList {
        id: String
        list: [Match]
        cursor: Float
    }

    type DateItem {
        id: String
        creator: User
        winner: User
        creationTime: String
        datetimeOfDate: String
        description: String
        bids: DateBidList
        num_bids: Int
        open: Boolean
        creationTimeEpoch: Float
        order: Float
    }

    type DateBid {
        id: String
        datetimeOfBid: String
        bidDescription: String
        bidPlace: String
        dateUser: User
        bidUser: User
        date: DateItem
    }

    type MessageItem {
        name: String
        avatar: String
        _id: String
        createdAt: String
        text: String
        order: Float
        uid: String
    }
    
    type Message {
        id: String 
        cursor: String
        list: [MessageItem]!
    }

    type Match {
        id: String
        matchId: String
        user: User
        messages: Message
        lastMessage: MessageItem
    }

    type Query {
        user(id: String!, hostId: String): User
        messages(id: String!): Message
        date(id: String!): DateItem
        dates(id: String!): DateList
        otherBids(id: String!): DateBidList
        moreMessages(id: String!, cursor: String!): Message
        moreQueue(id: String!, followerDisplay: String, cursor: Float!): Queue
        moreDates(id: String!, cursor: Float!): DateList
        moreDateBids(id:String!, cursor: Float!): DateBidList
        moreFollowing(id: String!, cursor: Float!): Following
    }

    type Subscription {
        newMessageSub(matchId: String, id: String): MessageItem
    }

    type Mutation {
        editUser (
            id: String!
            name: String
            active: Boolean
            email: String
            gender: String
            age: Int
            description: String
            school: String
            work: String
            sendNotifications: Boolean
            distance: Int
            token: String
            latitude: Float
            longitude: Float
            registerDateTime: String
            minAgePreference: Int
            maxAgePreference: Int
            followerDisplay: String
            pics: [String]
        ): User
        editUserQueue (
            id: String!
            sendNotifications: Boolean
            distance: Int
            minAgePreference: Int
            maxAgePreference: Int
            followPreference: Int
        ): Queue
        newUser (
            id: String!
            name: String
            active: Boolean
            email: String
            gender: String
            age: Int
            description: String
            school: String
            work: String
            sendNotifications: Boolean
            distance: Int
            token: String
            latitude: Float
            longitude: Float
            registerDateTime: String
            minAgePreference: Int
            maxAgePreference: Int
            followerDisplay: String
            pics: [String]
        ): User
        newMessage (
            matchId: String! 
            name: String 
            text: String
            createdAt: String
            avatar: String
            order: Float
            uid: String
            _id: String
            receiverId: String
        ): MessageItem
        follow (id: String!, followId: String!, isFollowing: Boolean): User
        unFollow (id: String!, unFollowId: String!): User
        bid(id: String!, dateId: String!, bidPlace: String, bidDescription: String): DateBid
        createDate(id: String!, datetimeOfDate: String, description: String): DateItem
        chooseWinner(id: String!, winnerId: String!, dateId: String!): DateItem
    }
    schema {
        query: Query
        mutation: Mutation
        subscription: Subscription
    }
`;

export default typeDefs;