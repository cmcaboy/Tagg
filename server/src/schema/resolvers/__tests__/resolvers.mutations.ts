import { server } from '../../schema';
import { newUserDefaults } from '../../defaults';
import {
  NEW_USER,
  NEW_DATE,
  BID,
  SET_COORDS,
  CHOOSE_WINNER,
  SET_PUSH_TOKEN,
  FOLLOW,
  UNFOLLOW,
  SEND_MESSAGE,
  SET_AGE_PREFERENCE,
  REMOVE_USER,
  SET_DISTANCE,
  SET_NOTIFICATIONS,
  SET_FOLLOWER_DISPLAY,
  SET_NAME,
  SET_AGE,
  SET_WORK,
  SET_SCHOOL,
  SET_DESCRIPTION,
  SET_PICS,
  SET_EMAIL,
} from '../../../clientGQL/mutation';

const moment = require('moment');

const { createTestClient } = require('apollo-server-testing');

const { ApolloServer } = require('apollo-server-express');

const TEST_ID = 'cory.mcaboy@gmail.com';
let DATE_ID = '';

const TEST_MESSAGE = {
  name: 'Cory McAboy',
  text: 'Hey, how are you?',
  createdAt: '10/01/2019',
  avatar:
    'https://firebasestorage.googleapis.com/v0/b/manhattanmatch-9f9fe.appspot.com/o/0a7a7fdb-39d9-484d-9b24-80fa0ef66240?alt=media&token=a79482d1-44f5-46be-87a2-56bce9340b89',
  order: 189342789,
  uid: 'testUser@test.com',
  _id: '3429048jfksdjlk342',
};

const TEST_USER = {
  ...newUserDefaults,
  id: 'testuser@test.com',
  name: 'John Smith',
  email: 'testuser@test.com',
  token: '1234567890abc',
  pics: [
    'https://firebasestorage.googleapis.com/v0/b/manhattanmatch-9f9fe.appspot.com/o/0a7a7fdb-39d9-484d-9b24-80fa0ef66240?alt=media&token=a79482d1-44f5-46be-87a2-56bce9340b89',
    'https://firebasestorage.googleapis.com/v0/b/manhattanmatch-9f9fe.appspot.com/o/04f85f3e-63f5-4231-89dd-a45f94e96fa7?alt=media&token=6de58f19-1b2d-4041-8451-beff7287bf38',
    'https://firebasestorage.googleapis.com/v0/b/manhattanmatch-9f9fe.appspot.com/o/0522af5a-8f39-42b1-a1ca-6e7b3e5eaf87?alt=media&token=e9bfda10-0600-41d2-a848-35a19cc26ec8',
    'https://firebasestorage.googleapis.com/v0/b/manhattanmatch-9f9fe.appspot.com/o/0a7a7fdb-39d9-484d-9b24-80fa0ef66240?alt=media&token=a79482d1-44f5-46be-87a2-56bce9340b89',
    'https://firebasestorage.googleapis.com/v0/b/manhattanmatch-9f9fe.appspot.com/o/0415e065-c80c-44b5-b469-4c6370804423?alt=media&token=86cf1038-8d6e-425c-90ef-b80ee999d29b',
    'https://firebasestorage.googleapis.com/v0/b/manhattanmatch-9f9fe.appspot.com/o/0a7a7fdb-39d9-484d-9b24-80fa0ef66240?alt=media&token=a79482d1-44f5-46be-87a2-56bce9340b89',
  ],
};

describe('Mutation ', () => {
  const testServer = new ApolloServer({
    ...server,
    context: () => ({
      user: {
        id: TEST_ID,
        email: TEST_ID,
      },
    }),
    engine: null,
  });
  const { mutate } = createTestClient(testServer);

  it('NEW_USER', async () => {
    const res = await mutate({ mutation: NEW_USER, variables: { ...TEST_USER } });
    expect(res).toMatchSnapshot();
  });
  it('NEW_DATE', async () => {
    const datetimeOfDate = moment().unix() + 10000;
    const description = 'Looking for a walk of the beach';

    const res = await mutate({
      mutation: NEW_DATE,
      variables: {
        id: TEST_USER.id,
        datetimeOfDate,
        description,
      },
    });
    // console.log('NEW_DATE res: ', res);
    // take record of DATE_ID for future use
    // console.log('res: ', res);
    DATE_ID = res.data.createDate.id;
    expect(res.data.createDate).toMatchSnapshot({
      creationTime: expect.any(Number),
      datetimeOfDate: expect.any(Number),
      description,
      id: expect.any(String),
    });
    expect(res.data.createDate.datetimeOfDate).toEqual(datetimeOfDate);
    expect(res.data.createDate.description).toEqual(description);
    // console.log('DATE_ID NEW_DATE: ', res.data.createDate.id);
  });
  it('BID', async () => {
    const bidDescription = 'Looking for a walk of the beach';
    const bidPlace = 'Harrys pub';

    const res = await mutate({
      mutation: BID,
      variables: {
        id: TEST_ID,
        dateId: DATE_ID,
        bidPlace,
        bidDescription,
      },
    });
    expect(res.data.bid).toMatchSnapshot({
      bidDescription,
      bidPlace,
      date: {
        id: expect.any(String),
        num_bids: expect.any(Number),
      },
      datetimeOfBid: expect.any(Number),
      id: expect.any(String),
    });
  });
  it('SET_COORDS', async () => {
    const res = await mutate({
      mutation: SET_COORDS,
      variables: {
        id: TEST_USER.id,
        latitude: 85.0,
        longitude: -35.0,
      },
    });
    expect(res).toMatchSnapshot();
  });
  it('SET_PUSH_TOKEN', async () => {
    const res = await mutate({
      mutation: SET_PUSH_TOKEN,
      variables: {
        id: TEST_USER.id,
        token: '12345',
      },
    });
    expect(res).toMatchSnapshot();
  });
  it('CHOOSE_WINNER', async () => {
    const res = await mutate({
      mutation: CHOOSE_WINNER,
      variables: {
        id: TEST_USER.id,
        winnerId: TEST_ID,
        dateId: DATE_ID,
      },
    });
    expect(res.data.chooseWinner).toMatchSnapshot({
      id: expect.any(String),
      datetimeOfDate: expect.any(Number),
      matchId: expect.any(String),
      lastMessage: null,
      user: expect.any(Object),
    });
    expect(res.data.chooseWinner.matchId).toBe(DATE_ID);
    expect(res.data.chooseWinner.id).toBe(DATE_ID);
    expect(res.data.chooseWinner.user.id).toBe(TEST_ID);
  });
  it('FOLLOW', async () => {
    const res = await mutate({
      mutation: FOLLOW,
      variables: {
        id: TEST_USER.id,
        followId: TEST_ID,
        isFollowing: true,
      },
    });
    expect(res).toMatchSnapshot();
  });
  it('UNFOLLOW', async () => {
    const res = await mutate({
      mutation: UNFOLLOW,
      variables: {
        id: TEST_USER.id,
        unFollowId: TEST_ID,
      },
    });
    expect(res).toMatchSnapshot();
  });
  it('SEND_MESSAGE', async () => {
    const receiverId = 'cory.mcaboy@gmail.com';

    const res = await mutate({
      mutation: SEND_MESSAGE,
      variables: {
        id: TEST_USER.id,
        matchId: DATE_ID,
        ...TEST_MESSAGE,
        receiverId,
      },
    });
    expect(res.data.newMessage).toMatchSnapshot({
      ...TEST_MESSAGE,
      createdAt: expect.any(String),
    });
  });
  it('SET_AGE_PREFERENCE', async () => {
    const NEW_MIN_AGE_PREF = 24;
    const NEW_MAX_AGE_PREF = 28;

    const res = await mutate({
      mutation: SET_AGE_PREFERENCE,
      variables: {
        id: TEST_USER.id,
        minAgePreference: NEW_MIN_AGE_PREF,
        maxAgePreference: NEW_MAX_AGE_PREF,
      },
    });
    expect(res).toMatchSnapshot();
    expect(res.data.editUser.minAgePreference).toEqual(NEW_MIN_AGE_PREF);
    expect(res.data.editUser.maxAgePreference).toEqual(NEW_MAX_AGE_PREF);
  });
  // SET_DISTANCE
  it('SET_DISTANCE', async () => {
    const DISTANCE = 45;

    const res = await mutate({
      mutation: SET_DISTANCE,
      variables: {
        id: TEST_USER.id,
        distance: DISTANCE,
      },
    });
    expect(res).toMatchSnapshot();
    expect(res.data.editUser.distance).toEqual(DISTANCE);
  });
  // SET_NOTIFICATIONS
  it('SET_NOTIFICATIONS', async () => {
    const NOTIFICATIONS = true;

    const res = await mutate({
      mutation: SET_NOTIFICATIONS,
      variables: {
        id: TEST_USER.id,
        sendNotifications: NOTIFICATIONS,
      },
    });
    expect(res).toMatchSnapshot();
    expect(res.data.editUser.sendNotifications).toEqual(NOTIFICATIONS);
  });
  // SET_FOLLOWER_DISPLAY
  it('SET_FOLLOWER_DISPLAY', async () => {
    const FOLLOWER_DISPLAY_BOTH = 'Both';

    const res = await mutate({
      mutation: SET_FOLLOWER_DISPLAY,
      variables: {
        id: TEST_USER.id,
        followerDisplay: FOLLOWER_DISPLAY_BOTH,
      },
    });
    expect(res).toMatchSnapshot();
    expect(res.data.editUser.followerDisplay).toEqual(FOLLOWER_DISPLAY_BOTH);
  });
  // SET_NAME
  it('SET_NAME', async () => {
    const NEW_NAME = 'Michael Smith';

    const res = await mutate({
      mutation: SET_NAME,
      variables: {
        id: TEST_USER.id,
        name: NEW_NAME,
      },
    });
    expect(res).toMatchSnapshot();
    expect(res.data.editUser.name).toEqual(NEW_NAME);
  });
  // SET_AGE
  it('SET_AGE', async () => {
    const NEW_AGE = 33;

    const res = await mutate({
      mutation: SET_AGE,
      variables: {
        id: TEST_USER.id,
        age: NEW_AGE,
      },
    });
    expect(res).toMatchSnapshot();
    expect(res.data.editUser.age).toEqual(NEW_AGE);
  });
  // SET_WORK
  it('SET_WORK', async () => {
    const NEW_WORK = 'Poppa Johns';

    const res = await mutate({
      mutation: SET_WORK,
      variables: {
        id: TEST_USER.id,
        work: NEW_WORK,
      },
    });
    expect(res).toMatchSnapshot();
    expect(res.data.editUser.work).toEqual(NEW_WORK);
  });
  // SET_SCHOOL
  it('SET_SCHOOL', async () => {
    const NEW_SCHOOL = 'Shawnee High School';

    const res = await mutate({
      mutation: SET_SCHOOL,
      variables: {
        id: TEST_USER.id,
        school: NEW_SCHOOL,
      },
    });
    expect(res).toMatchSnapshot();
    expect(res.data.editUser.school).toEqual(NEW_SCHOOL);
  });
  // SET_DESCRIPTION
  it('SET_DESCRIPTION', async () => {
    const NEW_DESCRIPTION = 'Looking for something special';

    const res = await mutate({
      mutation: SET_DESCRIPTION,
      variables: {
        id: TEST_USER.id,
        description: NEW_DESCRIPTION,
      },
    });
    expect(res).toMatchSnapshot();
    expect(res.data.editUser.description).toEqual(NEW_DESCRIPTION);
  });
  // SET_PICS
  it('SET_PICS', async () => {
    const NEW_PICS = [
      'https://firebasestorage.googleapis.com/v0/b/manhattanmatch-9f9fe.appspot.com/o/0a7a7fdb-39d9-484d-9b24-80fa0ef66240?alt=media&token=a79482d1-44f5-46be-87a2-56bce9340b89',
    ];

    const res = await mutate({
      mutation: SET_PICS,
      variables: {
        id: TEST_USER.id,
        pics: NEW_PICS,
      },
    });
    expect(res).toMatchSnapshot();
    expect(res.data.editUser.pics).toEqual(NEW_PICS);
  });
  // SET_EMAIL
  it('SET_EMAIL', async () => {
    const NEW_EMAIL = 'test2@test.com';

    const res = await mutate({
      mutation: SET_EMAIL,
      variables: {
        id: TEST_USER.id,
        email: NEW_EMAIL,
      },
    });
    expect(res.data.editUser.email).toEqual(NEW_EMAIL);
    expect(res).toMatchSnapshot();
    const reset = await mutate({
      mutation: SET_EMAIL,
      variables: {
        id: TEST_USER.id,
        email: TEST_USER.email,
      },
    });
    expect(reset.data.editUser.email).toEqual(TEST_USER.email);
  });

  it('REMOVE_USER', async () => {
    const res = await mutate({
      mutation: REMOVE_USER,
      variables: {
        id: TEST_USER.id,
      },
    });
    expect(res).toMatchSnapshot();
    expect(res.data.removeUser.id).toEqual(TEST_USER.id);
  });
});
