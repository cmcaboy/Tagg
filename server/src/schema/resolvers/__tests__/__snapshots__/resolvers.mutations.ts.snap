// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mutation  BID 1`] = `
Object {
  "bidDescription": "Looking for a walk of the beach",
  "bidPlace": "Harrys pub",
  "date": Object {
    "id": Any<String>,
    "num_bids": Any<Number>,
  },
  "datetimeOfBid": Any<Number>,
  "id": Any<String>,
}
`;

exports[`Mutation  CHOOSE_WINNER 1`] = `
Object {
  "datetimeOfDate": Any<Number>,
  "id": Any<String>,
  "lastMessage": null,
  "matchId": Any<String>,
  "user": Any<Object>,
}
`;

exports[`Mutation  FOLLOW 1`] = `
Object {
  "data": Object {
    "follow": Object {
      "id": "cory.mcaboy@gmail.com",
      "isFollowing": true,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Mutation  NEW_DATE 1`] = `
Object {
  "creationTime": Any<Number>,
  "datetimeOfDate": Any<Number>,
  "description": "Looking for a walk of the beach",
  "id": Any<String>,
}
`;

exports[`Mutation  NEW_USER 1`] = `
Object {
  "data": Object {
    "newUser": Object {
      "email": "testuser@test.com",
      "id": "testuser@test.com",
      "name": "John Smith",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Mutation  REMOVE_USER 1`] = `
Object {
  "data": Object {
    "removeUser": Object {
      "id": "testuser@test.com",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Mutation  SEND_MESSAGE 1`] = `
Object {
  "_id": "3429048jfksdjlk342",
  "avatar": "https://firebasestorage.googleapis.com/v0/b/manhattanmatch-9f9fe.appspot.com/o/0a7a7fdb-39d9-484d-9b24-80fa0ef66240?alt=media&token=a79482d1-44f5-46be-87a2-56bce9340b89",
  "createdAt": Any<String>,
  "name": "Cory McAboy",
  "order": 189342789,
  "text": "Hey, how are you?",
  "uid": "testUser@test.com",
}
`;

exports[`Mutation  SET_AGE 1`] = `
Object {
  "data": Object {
    "editUser": Object {
      "age": 33,
      "id": "testuser@test.com",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Mutation  SET_AGE_PREFERENCE 1`] = `
Object {
  "data": Object {
    "editUser": Object {
      "id": "testuser@test.com",
      "maxAgePreference": 28,
      "minAgePreference": 24,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Mutation  SET_COORDS 1`] = `
Object {
  "data": Object {
    "editUser": Object {
      "id": "testuser@test.com",
      "latitude": 85,
      "longitude": -35,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Mutation  SET_DESCRIPTION 1`] = `
Object {
  "data": Object {
    "editUser": Object {
      "description": "Looking for something special",
      "id": "testuser@test.com",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Mutation  SET_DISTANCE 1`] = `
Object {
  "data": Object {
    "editUser": Object {
      "distance": 45,
      "id": "testuser@test.com",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Mutation  SET_EMAIL 1`] = `
Object {
  "data": Object {
    "editUser": Object {
      "email": "test2@test.com",
      "id": "testuser@test.com",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Mutation  SET_FOLLOWER_DISPLAY 1`] = `
Object {
  "data": Object {
    "editUser": Object {
      "followerDisplay": "Both",
      "id": "testuser@test.com",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Mutation  SET_NAME 1`] = `
Object {
  "data": Object {
    "editUser": Object {
      "id": "testuser@test.com",
      "name": "Michael Smith",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Mutation  SET_NOTIFICATIONS 1`] = `
Object {
  "data": Object {
    "editUser": Object {
      "id": "testuser@test.com",
      "sendNotifications": true,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Mutation  SET_PICS 1`] = `
Object {
  "data": Object {
    "editUser": Object {
      "id": "testuser@test.com",
      "pics": Array [
        "https://firebasestorage.googleapis.com/v0/b/manhattanmatch-9f9fe.appspot.com/o/0a7a7fdb-39d9-484d-9b24-80fa0ef66240?alt=media&token=a79482d1-44f5-46be-87a2-56bce9340b89",
      ],
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Mutation  SET_PUSH_TOKEN 1`] = `
Object {
  "data": Object {
    "editUser": Object {
      "id": "testuser@test.com",
      "token": "12345",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Mutation  SET_SCHOOL 1`] = `
Object {
  "data": Object {
    "editUser": Object {
      "id": "testuser@test.com",
      "school": "Shawnee High School",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Mutation  SET_WORK 1`] = `
Object {
  "data": Object {
    "editUser": Object {
      "id": "testuser@test.com",
      "work": "Poppa Johns",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Mutation  UNFOLLOW 1`] = `
Object {
  "data": Object {
    "unFollow": Object {
      "id": "cory.mcaboy@gmail.com",
      "isFollowing": false,
      "name": "Jeff Smith",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
